### PR TITLE
Minor location tweaks

### PR DESCRIPTION
**Pull request by ltratt**

### CODE REVIEW

File: ykrt/src/mt.rs [hunk 1]
```diff
@@ -237,7 +237,11 @@ impl MT {
                     }
                     match loc.lock() {
                         Ok(x) => ls = x,
-                        Err(()) => unreachable!(),
+                        Err(()) => {
```
Comment (vext01) [inline ykrt/src/mt.rs:240]: What prompted this change? Did you encounter a crash? If so, do we have a test?

Comment (ltratt) [reply ykrt/src/mt.rs:240]: When reading the code, I realised this is reachable. Testing this sort of nondeterministic code exhaustively is impossible, unfortunately.

Comment (vext01) [reply ykrt/src/mt.rs:240]: ok

**Comment by vext01** [comment]
Any squashing to do? Probably not?

**Comment by ltratt** [comment]
None needed!

**Pull request merged**